code/implementation/docker/common/src/main/java/io/cattle/platform/docker/transform/DockerTransformerImpl.java

```diff
@@ -203,6 +211,71 @@ public void transform(Map<String, Object> fromInspect, Instance instance) {
         // Consider: AttachStdin, AttachStdout, AttachStderr, StdinOnce,
     }
 
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    void setBlkioDeviceOptionss(Instance instance, Map<String, Object> fromInspect) {
+        /*
+         * We're coverting from docker's structure of:
+         *  {BlkioDeviceReadIOps: [{Path: <device>, Rate: 1000} ... ], BlkioDeviceWriteIOps: [{Path: <device>, Rate: 1000} ... ] ... }
+         * to cattle's structure of:
+         *  {blkioDeviceOptions: {<device>: {readIops: 1000m writeIops: 1000 ... } ... }
+         */
+        List<String> fields = Arrays.asList(READ_IOPS, WRITE_IOPS, READ_BPS, WRITE_BPS, WEIGHT);
+        Map<String, BlkioDeviceOption> target = new HashMap<>();
+
+        for (String field : fields) {
+            List<Map> deviceOptions = null;
+            try {
+                deviceOptions = (List<Map>)CollectionUtils.toList(CollectionUtils.getNestedValue(fromInspect, "HostConfig", field));
```

Comment:
Nit: is HostConfig a constant anywhere? So we can reference it here instead of hardcoding